src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const versionsOfState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const KeyToRemove of action.keysToRemove) {
```

Comment:
Classes or constructors are called with an uppercase letter, and ordinary variables is usually called with a lowercase letter
```suggestion
        for (const keyToRemove of action.keysToRemove) {
```